src/pages/config.tsx

```diff
@@ -136,34 +137,67 @@ function ConfigPage (): React.JSX.Element | null {
     setResetKey((prev) => prev + 1)
   }, [])
 
-  const newlineStringSave = (value: string): string => JSON.stringify(convertUrlInputToArray(value))
-  const newlineStringLoad = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
+  const newlineToJsonArray = (value: string): string => JSON.stringify(convertUrlInputToArray(value))
+  const jsonToNewlineString = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
 
   return (
     <>
     {!isLoadedInIframe && <Header /> }
     <main className='e2e-config-page pa4-l bg-snow mw7 center pa4'>
       <Collapsible collapsedLabel="View config" expandedLabel='Hide config' collapsed={isLoadedInIframe}>
+      <LocalStorageToggle
+        className="e2e-config-page-input"
+        label="Enable Recursive Gateways"
+        description="Use recursive gateways configured below for retrieval of content."
+        defaultValue={defaultEnableRecursiveGateways}
+        localStorageKey={LOCAL_STORAGE_KEYS.config.enableRecursiveGateways}
+        resetKey={resetKey}
+      />
         <LocalStorageInput
           className="e2e-config-page-input e2e-config-page-input-gateways"
-          description="A newline delimited list of trustless gateway URLs."
+          description="A newline delimited list of recursive trustless gateway URLs."
           localStorageKey={LOCAL_STORAGE_KEYS.config.gateways}
-          label='Gateways'
+          label='Recursive Gateways'
```

Comment:
A gateway could be recursive without being trustless, yes? But a recursive, non-trustless, gateway wouldn't work for verified-fetch calls

```suggestion
          label='Trustless Gateways'
```

Reply:
It's mostly to make a distinction from **non-recursive** gateway providers.

Recursive is here to indicate that the gateway is handling routing/retrieval for you.

Reply:
I guess that's true, but are all trustless gateways recursive? maybe we can just add that in the description?

Reply:
No. The classic example is w3link which only returns blocks it has. Almost all the gateways coming from the ipni are non recursive and essential just serve cached blocks over a trustless gateway